### ISSUE TITLE
ext_authz: allow updating path querystring when allowing request

### ISSUE THREAD

**Issue by erikbos** [envoyproxy/envoy#10089]
*Description*:
I'd love to have the capability to update/rewrite the query string of the request in an ext_authz response before it is forwarded upstream.

This feature would be useful in scenarios where it is required to strip sensitive information from the request before forwarding it upstream. An example use case is APIs where apikey is provided as query parameter.

e.g.
1) client retrieves http://api/users/1?apikey=abc&format=json

2) ext_authz checks key abc in database and allows request

3) ext_authz responds with updated path/query string without apikey=abc

4) upstream retrieves request for /users/1?format=json

(My main need is to to be able to update query parameter string before forwarding the request, but one could reason a more generic implementation could be to allow updating the full path)

**Comment by dio** [comment]
So this is similar to `allowed_upstream_headers` (e.g. to be able to state: I only allow `header-name` and `content-type` to be forwarded to upstream) but for query params?

**Comment by erikbos** [comment]
Indeed, you could see it like `allowed_upstream_headers` : a way to specify which query parameters are allowed to go upstream.

But.. such allowed list implementation is not ideal for the scenario described above: given I have tens of upstreams I'd rather not specify all possible query parameters supported by those upstreams.

It's more the opposite: in my case I'd like to be able to drop both query parameter `apikey` and `key` before the request goes upstream, as either can be used by my customers.

So for my use case I would be more looking for a feature like `disallowed_queryparameters` as part of AuthorizationResponse instead of `allowed_queryparameters`  (perhaps somebody has a usecase for `allowed_queryparameters`..)

Let me know if this makes sense!
